qbft/future_msg.go

```diff
@@ -0,0 +1,68 @@
+package qbft
+
+import (
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/pkg/errors"
+)
+
+func (c *Controller) UponFutureMsg(msg *SignedMessage) (*SignedMessage, error) {
+	if err := validateFutureMsg(c.GetConfig(), msg, c.Share.Committee); err != nil {
+		return nil, errors.Wrap(err, "invalid future msg")
+	}
+	if !c.addHigherHeightMsg(msg) {
+		return nil, errors.New("discarded future msg")
+	}
+	if c.f1SyncTrigger() {
+		return nil, c.GetConfig().GetNetwork().SyncHighestDecided(c.Identifier)
+	}
+	return nil, nil
+}
+
+func validateFutureMsg(
```

Comment:
Again, if you admit group membership changes, you can't really validate a future message because you don't know who the correct signers will be by then.